models/migrations/v1_19/v238.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_19 //nolint
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+// AddCardTypeToProjectTable: add CardType column, setting existing rows to CardTypeTextOnly
+func AddCardTypeToProjectTable(x *xorm.Engine) error {
+	type Project struct {
+		CardType int `xorm:"NOT NULL DEFAULT 1"`
```

Comment:
We won't need `DEFAULT 1` if `CardTypeTextOnly` is 0.

```suggestion
		CardType int `xorm:"NOT NULL"`
```

Reply:
Sorry, my bad, I overlooked `NOT NULL`.